libs/wallet/src/connect-dialog/connect-dialog.tsx

```diff
@@ -28,15 +29,37 @@ type WalletType = 'gui' | 'cli' | 'hosted';
 
 export interface VegaConnectDialogProps {
   connectors: Connectors;
+  onChangeOpen?: (open: boolean) => void;
+}
+
+export const useVegaWalletDialogStore = create<VegaWalletDialogStore>(
+  (set) => ({
+    dialogOpen: false,
+    updateDialogOpen: (open: boolean) => set({ dialogOpen: open }),
+  })
+);
+
+interface VegaWalletDialogStore {
   dialogOpen: boolean;
-  setDialogOpen: (isOpen: boolean) => void;
+  updateDialogOpen: (open: boolean) => void;
 }
 
 export const VegaConnectDialog = ({
   connectors,
-  dialogOpen,
-  setDialogOpen,
+  onChangeOpen,
```

Comment:
Why we cannot name it like in radix ui Dialog? 🤔
`open` and `onOpenChange`

Reply:
yeah, I realised the names of properties os the store are not quite good, but I thought rather about names which inform we are dealing with vegaWallet dialog. So what about names like `vegaWalletDialogOpen` and `updateVegaWalletDialog` ? What do you think?